lightclient/src/platform/wasm_socket.rs

```diff
@@ -171,6 +171,10 @@ impl AsyncRead for WasmSocket {
         let mut inner = self.inner.lock().expect("Mutex is poised; qed");
         inner.waker = Some(cx.waker().clone());
 
+        if self.socket.ready_state() == web_sys::WebSocket::CONNECTING {
+            return Poll::Pending;
```

Comment:
So I think this is ook because any of the callbacks will wake the waker when JS clals them, ensuring we call this `poll_read` again.